src/System.Net.Sockets/tests/FunctionalTests/UdpClientTest.cs

```diff
@@ -142,5 +458,214 @@ public void AllowNatTraversal_AnyUnix(bool allow)
                 Assert.Throws<PlatformNotSupportedException>(() => c.AllowNatTraversal(allow));
             }
         }
+
+        [OuterLoop] // TODO: Issue #11345
+        [ConditionalTheory(nameof(PlatformDetection) + "." + nameof(PlatformDetection.IsNotWindowsSubsystemForLinux))] // https://github.com/Microsoft/BashOnWindows/issues/987
+        [InlineData(false)]
+        [InlineData(true)]
+        public void Send_Receive_Success(bool ipv4)
+        {
+            IPAddress address = ipv4 ? IPAddress.Loopback : IPAddress.IPv6Loopback;
+
+            using (var receiver = new UdpClient(new IPEndPoint(address, 0)))
+            using (var sender = new UdpClient(new IPEndPoint(address, 0)))
+            {
+                for (int i = 0; i < TestSettings.UDPRedundancy; i++)
+                {
+                    sender.Send(new byte[1], 1, new IPEndPoint(address, ((IPEndPoint)receiver.Client.LocalEndPoint).Port));
+                }
+
+                IPEndPoint remoteEP = null;
+                byte[] data = receiver.Receive(ref remoteEP);
+                Assert.NotNull(remoteEP);
+                Assert.InRange(data.Length, 1, int.MaxValue);
+            }
+        }
+
+        [PlatformSpecific(TestPlatforms.Windows)] // "localhost" resolves to IPv4 & IPV6 on Windows, but may resolve to only one of those on Unix 
```

Comment:
Because a UdpClient instance is associated with a single AddressFamily, defaulting to InterNetwork (IPv4); if "localhost" ends up resolving to IPv6 only, then it fails.